crates/holochain_zome_types/src/zome_io.rs

```diff
@@ -190,10 +210,77 @@ pub enum ZomeCallResponse {
     Ok(crate::ExternIO),
     /// Cap grant failure.
     /// Something like a 401 http response.
-    Unauthorized(CellId, ZomeName, FunctionName, AgentPubKey),
+    Unauthorized(
+        ZomeCallAuthorization,
+        CellId,
+        ZomeName,
+        FunctionName,
+        AgentPubKey,
+    ),
     /// This was a zome call made remotely but
     /// something has failed on the network
     NetworkError(String),
     /// A countersigning session has failed to start.
     CountersigningSession(String),
 }
+
+/// 256 Bit generic nonce.
+#[derive(Clone, Deserialize, Serialize)]
+pub struct Nonce256Bits([u8; 32]);
```

Comment:
```suggestion
pub struct Nonce256Bits(#[serde(with = "serde_bytes")] [u8; 32]);
```

Reply:
@neonphog When I'm adding this attribute now, I get the error that `cannot find attribute `serde` in this scope
`serde` is in scope, but it is a crate, not an attribute`. How can I fix that?

Reply:
@jost-s - very strange, I don't have any immediate thoughts. It looks identical to this usage: [https://github.com/holochain/holochain/blob/develop/crates/kitsune_p2p/kitsune_p2p/src/types/wire.rs#L14](https://github.com/holochain/holochain/blob/develop/crates/kitsune_p2p/kitsune_p2p/src/types/wire.rs#L14).

If you can't figure it out, you might have to manually derive serialize/deserialize to use the bytes variants.

Reply:
@jost-s - If that were true, there wouldn't be a `#[derive(Serialize)]` on this struct. That default derive is putting the values as integers into an array, rather than treating them as a byte slice.

Reply:
@neonphog Check the code in the other tab, the above version is outdated. https://github.com/holochain/holochain/pull/1510/files#diff-7a679accda04d72e8da35c1168acad067fa0f1f111ec75ac79c80a532be69340R228

Reply:
Ah. Ok, cool :+1: